be/src/runtime/types.cpp

```diff
@@ -270,6 +270,15 @@ TypeDescriptor::TypeDescriptor(const google::protobuf::RepeatedPtrField<PTypeNod
     }
 }
 
+void TypeDescriptor::add_sub_type(TypeDescriptor&& sub_type, std::string&& field_name,
+                                  bool&& is_nullable) {
+    children.emplace_back(sub_type);
+    if (!field_name.empty()) {
```

Comment:
Better use a new method add_field_name for field_name.
Because field_name is only used for struct type.